pkg/local_object_storage/metabase/lock.go

```diff
@@ -0,0 +1,159 @@
+package meta
+
+import (
+	"bytes"
+	"fmt"
+
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.etcd.io/bbolt"
+)
+
+// bucket name for locked objects.
+var bucketNameLocked = []byte(invalidBase58String + "Locked")
+
+// suffix for container buckets with objects of type LOCK.
+const bucketNameSuffixLockers = invalidBase58String + "LOCKER"
+
+// returns name of the bucket with objects of type LOCK for specified container.
+func bucketNameLockers(idCnr cid.ID) []byte {
+	return []byte(idCnr.String() + bucketNameSuffixLockers)
+}
+
+// Lock marks objects as locked with another object. All objects from the
+// specified container.
+//
+// Allows locking regular objects only (otherwise returns apistatus.LockNonRegularObject).
+//
+// Locked list should be unique. Panics if it is empty.
+func (db *DB) Lock(cnr cid.ID, locker oid.ID, locked []oid.ID) error {
+	if len(locked) == 0 {
+		panic("empty locked list")
+	}
+
+	return db.boltDB.Update(func(tx *bbolt.Tx) error {
+		// check if all objects are regular
+		bucketKeysLocked := make([][]byte, len(locked))
+
+		for i := range locked {
+			bucketKeysLocked[i] = objectKey(&locked[i])
+		}
+
+		if firstIrregularObjectType(tx, cnr, bucketKeysLocked...) != object.TypeRegular {
+			return apistatus.LockNonRegularObject{}
+		}
+
+		bucketLocked, err := tx.CreateBucketIfNotExists(bucketNameLocked)
+		if err != nil {
+			return fmt.Errorf("create global bucket for locked objects: %w", err)
+		}
+
+		bucketLockedContainer, err := bucketLocked.CreateBucketIfNotExists([]byte(cnr.String()))
+		if err != nil {
+			return fmt.Errorf("create container bucket for locked objects %v: %w", cnr, err)
+		}
+
+		keyLocker := objectKey(&locker)
+		var exLockers [][]byte
+		var updLockers []byte
+
+	loop:
+		for i := range bucketKeysLocked {
+			// decode list of already existing lockers
+			exLockers, err = decodeList(bucketLockedContainer.Get(bucketKeysLocked[i]))
+			if err != nil {
+				// maybe continue for the best effort?
```

Comment:
discuss/TODO/issue?

Reply:
We will consider this question someday, now we're definitely not ready.

Reply:
create issue in the `triage`? afraid it could be forgotten

Reply:
Removed the comment.